pycoal/tests/mineral_test.py

```diff
@@ -112,3 +112,61 @@ def test_toRGB_AVC():
     assert expected.metadata.get('fwhm') == actual.metadata.get('fwhm')
     assert expected.metadata.get('bbl') == actual.metadata.get('bbl')
     assert expected.metadata.get('smoothing factors') == actual.metadata.get('smoothing factors')
+
+# test files for the classifyImage test
+#TODO use an AVIRIS-NG image for two test cases
+test_classifyImage_testFilename_1 = "pycoal/tests/ang20140912t192359_corr_v1c_img_400-410_10-20.hdr"
+test_classifyImage_testFilename_2 = "pycoal/tests/ang20140912t192359_corr_v1c_img_2580-2590_540-550.hdr"
+test_classifyImage_testFilename_3 = "pycoal/tests/ang20150422t163638_corr_v1e_img_4000-4010_550-560.hdr"
+
+ftp_url = "ftp://ftpext.cr.usgs.gov/"
+library_files = ["s06av95a_envi.hdr", "s06av95a_envi.sli"]
+
+# delete temporary files
+def _test_classifyImage_teardown():
+    files = [test_classifyImage_testFilename_1[:-4] + "_class.hdr",
+             test_classifyImage_testFilename_1[:-4] + "_class.img",
+             test_classifyImage_testFilename_2[:-4] + "_class.hdr",
+             test_classifyImage_testFilename_2[:-4] + "_class.img",
+             test_classifyImage_testFilename_3[:-4] + "_class.hdr",
+             test_classifyImage_testFilename_3[:-4] + "_class.img",
+             "pycoal/tests/" + library_files[0],
+             "pycoal/tests/" + library_files[1]]
+
+    for f in files:
+        try:
+            os.remove(f)
+        except OSError:
+            pass
+
+
+# verify that classified images have valid classifications
+@with_setup(None, _test_classifyImage_teardown)
+def test_classifyImage():
+    ftp = ftplib.FTP("ftpext.cr.usgs.gov")
+    ftp.login()
+
+    ftp.cwd("pub/cr/co/denver/speclab/pub/spectral.library/splib06.library/Convolved.libraries/")
+
+    for f in library_files:
+        with open("pycoal/tests/" + f, "wb") as lib_f:
+            ftp.retrbinary('RETR %s' % f, lib_f.write)
+
+    lib = "pycoal/tests/s06av95a_envi.hdr"
+
+    tst_cls = mineral.MineralClassification(lib)
+
+    # make sure library is being opened as such
+    assert isinstance(tst_cls.library, spectral.io.envi.SpectralLibrary)
+
+    for f in (test_classifyImage_testFilename_1, test_classifyImage_testFilename_2, test_classifyImage_testFilename_3):
+        tst_cls.classifyImage(f, f[:-4] + "_class.hdr")
+
+        classified = spectral.open_image(f[:-4] + "_class.hdr")
+
+        assert classified.metadata.get(u'description') == u'PyCOAL 0.2 mineral classified image.'
+        assert classified.metadata.get(u'file type') == u'ENVI Classification'
+
+        # assert there are no invalid class numbers
+        for i in classified.asarray().flatten():
+            assert 0 <= i <= len(tst_cls.library.names) + 1
```

Comment:
as we have discussed, an additional thing we could test is that the class of each pixel equals the value in some test expected test subimage.